### PR TITLE
Intellisense fixes

### LINKED ISSUE #370: Reverse expression parser needs updating for async
```python
async def bar():
    return 100

async def foo():
    (await bar())
```

Do . after (await bar()) and you get generator completions instead of int completions.

### PR DESCRIPTION
**Pull request by zooba**
#370 Reverse expression parser needs updating for async
Always parses context-dependent keywords as keywords when parsing isolated expressions.
Improves stability of tests.

Fixes tokenizer crash when parsing invalid complex value.

#374 Local variables are not initially colorized
Ensures the event is hooked and classifications are calculated when analysis is complete before classifications are requested.
Improves test stability
(Originally wasn't going to look at #374 for 2.2, but needed the fix for test stability or we wouldn't ever get classifications.)

### CODE REVIEW

File: Python/Product/Analysis/PythonKeywords.cs [hunk 1]
```diff
@@ -72,15 +72,15 @@ public static class PythonKeywords {
         public static IEnumerable<string> Expression(PythonLanguageVersion version = PythonLanguageVersion.None) {
             yield return "and";
             yield return "as";
-            if (version >= PythonLanguageVersion.V35) {
+            if (version == PythonLanguageVersion.None || version >= PythonLanguageVersion.V35) {
```
Comment (gilbertw) [inline Python/Product/Analysis/PythonKeywords.cs]: version.IsNone()

Comment (zooba) [inline Python/Product/Analysis/PythonKeywords.cs]: As in, create a new function for this?

Comment (gilbertw) [inline Python/Product/Analysis/PythonKeywords.cs]: Yes,
static bool IsNone(this PythonLanguageVersion version){
  return version == PythonLanguageVersion.None;
}

**Comment by int19h** [comment]
:+1:

**Comment by crwilcox** [comment]
:+1:  lgtm

**Pull request merged**